### ISSUE TITLE
option to use an existing certificate

### ISSUE THREAD

**Issue by lexelby** [GoogleCloudPlatform/terraform-google-lb-http#10]
We would find it useful to be able to pass in the `self_link` of an existing certificate to `lb-http`.  This would allow us to create the certificate outside of terraform to avoid having key material in the tfstate file.

I've implemented this feature in our fork: https://github.com/fastly/terraform-google-lb-http/pull/1

To my understanding, Fastly already has a corporate CLA, and I've been added to it.  If this all looks good to you, I'll create a PR.

**Comment by neomantra** [comment]
@lexelby Maybe make a pull request from that Fastly repo to this repo?

**Comment by lexelby** [comment]
Sure, I'd be happy to. But CONTRIBUTING.md says to open an issue and wait for a maintainer to ask me to submit a PR if they like the idea.

**Comment by neomantra** [comment]
Ah, my bad.  Thanks for sharing this.

**Comment by eltimn** [comment]
@lexelby if you don't mind me asking, how do you get a reference to the `self_link` of an existing certificate?

**Comment by lexelby** [comment]
@eltimn Like this:

```
gcloud compute ssl-certificates list --project <project ID> --uri
```

**Comment by eltimn** [comment]
Excellent. Thank you!

**Comment by lexelby** [comment]
@danisla @danawillow Code for this is working and tested in our environment.  How can I help you folks to get this moving?  My employer (Fastly) has signed a CLA that includes me, so that shouldn't be a stumbling block.

**Comment by danisla** [comment]
This is great, please create a PR and I'll get it merged.

**Comment by lexelby** [comment]
@danisla created: https://github.com/GoogleCloudPlatform/terraform-google-lb-http/pull/19

Interestingly, #18 seems to also have the same bring-your-own-cert functionality.

**Comment by danisla** [comment]
Fixed with #16
